apps/heft/src/cli/actions/CustomAction.ts

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import {
+  CommandLineFlagParameter,
+  CommandLineStringParameter,
+  CommandLineIntegerParameter,
+  CommandLineStringListParameter
+} from '@rushstack/ts-command-line';
+
+import { HeftActionBase, IHeftActionBaseOptions } from './HeftActionBase';
+
+/** @alpha */
+export interface ICustomActionParameterFlag extends ICustomActionParameterBase<boolean> {
+  kind: 'flag';
+}
+
+/** @alpha */
+export interface ICustomActionParameterInteger extends ICustomActionParameterBase<number> {
+  kind: 'integer';
+}
+
+/** @alpha */
+export interface ICustomActionParameterString extends ICustomActionParameterBase<string> {
+  kind: 'string';
+}
+
+/** @alpha */
+export interface ICustomActionParameterStringList extends ICustomActionParameterBase<ReadonlyArray<string>> {
+  kind: 'stringList';
+}
+
+/** @alpha */
```

Comment:
The Rush Stack repo doesn't have a support model that would meaningfully distinguish `@alpha` versus `@beta`.  Should probably do a PR to bulk search+replace `@alpha` with `@beta` everywhere.

We should probably also search for any APIs that have stagnated as `@beta` for more than a year and consider promoting them to `@public`.